Connector/btc/apirpc.py

```diff
@@ -393,53 +393,48 @@ def getTransaction(id, params, config):
         raise error.RpcBadRequestError(err.message)
 
     try:
-        # Parameters: TransactionId, include_watchonly, verbose
         transaction = RPCConnector.request(
             endpoint=config.bitcoincoreRpcEndpoint,
             id=id,
-            method=GET_TRANSACTION_METHOD,
+            method="getrawtransaction",
             params=[
                 params["txHash"],
-                True,
                 True
             ]
         )
 
-        vinAddressBalances = {}
-        transactionAmount = 0
+        isConfirmed = "blockhash" in transaction
+
+        if isConfirmed:
+            transactionBlock = RPCConnector.request(
+                endpoint=config.bitcoincoreRpcEndpoint,
+                id=id,
+                method="getblock",
+                params=[transaction["blockhash"], 1]
+            )
+            blockNumber = transactionBlock["height"]
+        else:
+            blockNumber = None
 
-        if "generated" not in transaction:
 
-            for vin in transaction["decoded"]["vin"]:
-                inputTransaction = RPCConnector.request(
-                    endpoint=config.bitcoincoreRpcEndpoint,
-                    id=id,
-                    method=GET_TRANSACTION_METHOD,
-                    params=[
-                        vin["txid"],
-                        True,
-                        True
-                    ]
-                )
+        transactionDetails = utils.decodeTransactionDetails(transaction, config.bitcoincoreRpcEndpoint)
 
-                transactionAmount += inputTransaction["decoded"]["vout"][vin["vout"]]["value"]
-                address = inputTransaction["decoded"]["vout"][vin["vout"]]["scriptPubKey"]["addresses"][0]
-                value = inputTransaction["decoded"]["vout"][vin["vout"]]["value"]
-                vinAddressBalances[address] = value
+        # Converting all transaction details to str
+        transactionDetails["fee"] = str(transactionDetails["fee"])
```

Comment:
No need to store the string value, just return the string value in response variable, "fee":str(transactionDetails["fee"])

Reply:
I disagree with this, because we are converting all elements in the array at once. I do not want to mix str variables with not-str ones in the same array.

Reply:
The thing is that the behaviour is the same if you delete this line and add `"fee":str(transactionDetails["fee"])` in line 434